rampwf/workflows/simplified_image_classifier.py

```diff
@@ -0,0 +1,161 @@
+from __future__ import division
+import os
+import imp
+import numpy as np
+from skimage.io import imread
+
+
+class SimplifiedImageClassifier(object):
+    """
+    SimplifiedImageClassifier workflow.
+    This workflow is used to train image classification tasks, typically when
+    the dataset cannot be stored in memory. It is a simplified version
+    of the `ImageClassifier` workflow where there is no batch generator
+    and no image preprocessor. 
+    Submissions need to contain one file, which by default by is named
+    batch_classifier.py (it can be modified by changing 
+    `workflow_element_names`).
+    batch_classifier.py needs a `BatchClassifier` class, which implements
+    `fit` and `predict_proba`, where both `fit` and `predict_proba` take 
+    as input an instance of `ImageLoader`.
+
+    Parameters
+    ==========
+
+    n_classes : int
+        Total number of classes.
+
+    """
+    def __init__(self, n_classes, workflow_element_names=['batch_classifier']):
```

Comment:
batch_classifier is a bit a strange name if there is no batch anymore

Reply:
Indeed, maybe we can just use image_classifier.

Reply:
I would do that, then with eg a ImageClassifier class the participants need to implement